### ISSUE TITLE
Extracting a mutator "quark" from a stack breaks the saved connections

### ISSUE THREAD

**Issue by BeksOmega** [google/blockly#6137]
**Describe the bug**

When you extract a mutator block from a stack of mutator blocks, by holding ctrl and then dragging, it disconnects all of the blocks associated with input, and doesn't properly reconnect them.

**To Reproduce**

1. Load the following blocks into the playground:
```
{
  "blocks": {
    "languageVersion": 0,
    "blocks": [
      {
        "type": "controls_if",
        "extraState": {
          "elseIfCount": 3
        },
        "inputs": {
          "IF0": {
            "block": {
              "type": "logic_boolean",
              "fields": {
                "BOOL": "TRUE"
              }
            }
          },
          "IF1": {
            "block": {
              "type": "logic_boolean",
              "fields": {
                "BOOL": "TRUE"
              }
            }
          },
          "IF2": {
            "block": {
              "type": "logic_boolean",
              "fields": {
                "BOOL": "TRUE"
              }
            }
          },
          "IF3": {
            "block": {
              "type": "logic_boolean",
              "fields": {
                "BOOL": "TRUE"
              }
            }
          }
        }
      }
    ]
  }
}
```
2. Open the mutator
3. Hold control
4. Drag the top if-else block out of the stack of mutator blocks.
5. Observe that the true/false blocks are not properly reconnected to their inputs.

**Expected behavior**

Blocks should always be properly reconnected to their inputs.

**Screenshots**

![Screen recording 2022-05-02 3 01 10 PM](https://user-images.githubusercontent.com/25440652/166334886-176a5722-cbdd-4e7a-9c3c-540083546190.gif)

**Additional context**

Reported originally: https://github.com/google/blockly/issues/6131#issuecomment-1113815160
Possibly related to: #6033

**Comment by tweini** [comment]
can reproduce on master
**can't reproduce on develop**

Guess #6047 has fixed it. In the fixed Issue #6033 you mentioned:
>Only tested with the above two blocks, not sure if it also occurs with if/else blocks.
Also reproduces on master.

So, you answered this question: Yes it occurs with if/else blocks.

But all seems to be fixed 😃

**Comment by BeksOmega** [comment]
> **can't reproduce on develop**

Haha nice! I was just testing on the hosted site. Thanks for trying to repro on develop :D

Closing this as cannot reproduce.
